### PR TITLE
Fix drive classes axes conventions

### LINKED ISSUE #2134: Axis conventions documentation incorrect for Mecanum Drive. Still states it is NED, rather than NWU as listed in changes for 2023.
The incorrect information is found on this page. https://docs.wpilib.org/en/latest/docs/software/hardware-apis/motors/wpi-drive-classes.html

### PR DESCRIPTION
**Pull request by rzblue**
A new image is needed, but I removed it for the time being.
Closes #2134

### CODE REVIEW

**Review by sciencewhiz — DISMISSED** [review]
The mecanum drive examples should need inversion updates

**Comment by Starlight220** [comment]
Assuming you're referring to the allwpilib examples, I think those did get updated to the correct coordinate frame.

**Comment by sciencewhiz** [comment]
There's code in the document that is not RLIed. That's what I was referring to.

File: source/docs/software/hardware-apis/motors/wpi-drive-classes.rst [hunk 1]
```diff
@@ -345,22 +340,22 @@ The MecanumDrive class contains two different default modes of driving your robo
     .. code-tab:: java
 
         public void teleopPeriodic() {
-            m_robotDrive.driveCartesian(-m_stick.getY(), m_stick.getX(), m_stick.getZ());
-            m_robotDrive.drivePolar(-m_stick.getY(), m_stick.getX(), m_stick.getZ());
+            m_robotDrive.driveCartesian(-m_stick.getY(), -m_stick.getX(), -m_stick.getZ());
+            m_robotDrive.drivePolar(-m_stick.getY(), -m_stick.getX(), -m_stick.getZ());
```
Comment (rzblue) [inline source/docs/software/hardware-apis/motors/wpi-drive-classes.rst:344]: This isn't really a correct usage of drivePolar anyway

Comment (rzblue) [reply source/docs/software/hardware-apis/motors/wpi-drive-classes.rst:344]: It's also not a valid function call- drivePolar requires a rotation2d. I'll try to fix this later.

Would converting the joystick "coordinates" to polar and using those be an acceptable demonstration, even though it's not a real world use case?

Comment (Starlight220) [reply source/docs/software/hardware-apis/motors/wpi-drive-classes.rst:344]: Is there a real world use case of `drivePolar`?

Comment (sciencewhiz) [reply source/docs/software/hardware-apis/motors/wpi-drive-classes.rst:344]: Polar might be easier to use in auto.

From usage reporting, 3 teams used drivePolar. However, a team that used polar in auto and cartesian in teleop I think would return cartesian in usage reporting.

Comment (Starlight220) [reply source/docs/software/hardware-apis/motors/wpi-drive-classes.rst:344]: Are there any uses of drivePolar in the example projects?

Comment (sciencewhiz) [reply source/docs/software/hardware-apis/motors/wpi-drive-classes.rst:344]: Reviewing uses of drivePolar across github, I didn't see any that used it from joysticks. Using it with hard coded angles seems to be how it's used

**Comment by rzblue** [comment]
I changed the example for drivepolar to be valid code, albeit not a common use case. Let me know how it looks.

**Review by Daltz333 — APPROVED** [review]

**Pull request merged**